### DISCUSSION TITLE
Entity

### DISCUSSION THREAD

**Discussion by Ilqjx** [SeaQL/sea-orm#538] [Q&A] [answered]
I use sea-orm-cli to generate entities, but I don't see Entity in the file.

pub use super::animal::Entity as Animal;

```rust
//! SeaORM Entity. Generated by sea-orm-codegen 0.6.0

use sea_orm::entity::prelude::*;

#[derive(Clone, Debug, PartialEq, DeriveEntityModel)]
#[sea_orm(table_name = "animal")]
pub struct Model {
    #[sea_orm(primary_key, auto_increment = false)]
    pub id: i32,
    pub name: String,
}

#[derive(Copy, Clone, Debug, EnumIter)]
pub enum Relation {}

impl RelationTrait for Relation {
    fn def(&self) -> RelationDef {
        panic!("No RelationDef")
    }
}

impl ActiveModelBehavior for ActiveModel {}

```

**Comment by billy1624** [comment, marked as answer]
Hey @Ilqjx, welcome! Good questions!

The `DeriveEntityModel` proc_macro actually defined `Entity` struct and implemented `EntityName` trait for you under the hood.

```rust
#[derive(Copy, Clone, Default, Debug, DeriveEntity)]
pub struct Entity;

impl EntityName for Entity {
    fn schema_name(&self) -> Option<&str> {
        None
    }

    fn table_name(&self) -> &str {
        "cake"
    }
}
```

_Adapted from https://www.sea-ql.org/SeaORM/docs/generate-entity/expanded-entity-structure#entity_

The source code of `DeriveEntityModel` proc_macro:

https://github.com/SeaQL/sea-orm/blob/56cf4ea662dc4865176650f34b0bc08cbed97999/sea-orm-macros/src/derives/entity_model.rs#L45-L59

**Comment by Ilqjx** [comment]
@billy1624 Thanks for your reply
